### DISCUSSION TITLE
I have 92 MD files, but Retype insists I need a Pro License!

### DISCUSSION THREAD

**Discussion by WitekioPhil** [retypeapp/retype#362] [Q&A] [answered]
I switched from paper-based notebooks to a digital notebook based on Retype nearly a year ago. It's a great product and I've been very happy with it.

I tried to open my notebook yesterday and encountered: "ERROR: Retype license is required."

I work in the SW industry, so I fully accept the need to at least cover costs and have no objection to the Pro offering in principle.

I am somewhat irked to have fewer pages than the 100-page limit for the free service, yet retype is demanding a license, with no useful information to help understand why, and therefore resolve (other than buying a key, which according to Retype's own docs, I shouldn't need 🤦).

I tried the verbose option but this adds no useful information:

`Retype started...
ERROR: Retype license is required.`

Perhaps you could improve this message by explaining WHY a retype license is required? Even better, include a listing of the files retype is counting towards this 100 page limit.

Any idea why Retype is refusing to launch without a Pro license when I only have 92 MD files?

![image](https://user-images.githubusercontent.com/78737454/178967945-df5a2ee9-0aa0-410f-a581-a1b0b193750c.png)

**Comment by geoffreymcgill** [comment, marked as answer]
Hi @WitekioPhil. Is your project available on GitHub or GitLab? Somewhere we could build and try to reproduce the issue.

If you run the command `retype watch` or `retype build`, you should get a list of basic stats showing the page count. Something like the following:

```
Retype finished in 2.5 seconds!
  46 pages built
   0 errors
   0 warnings
```

Can you share the stats from your project?

If you run `retype watch --verbose`, then you should get a mega list of all the pages and build steps taken by Retype.

I don't think the message `Retype license is required` is related to the 100 page count. There is likely something else going on that Retype is getting triggered on. I'll do some research to try come up with other possible causes, although being able to build your project locally would be get to the root of the issue quickest.

Hope this helps.

**Reply by geoffreymcgill** [reply]
Please run the command `retype wallet --list`. 

1. Do any keys get listed?
2. Are any listed with `- Status: Expired`?
3. Can you share the results of the `retype wallet --list` command?

**Reply by WitekioPhil** [reply]
Hi @geoffreymcgill!

Many thanks for the swift response.

It's a personal notebook I use for work notes, most of the content is under NDA of one sort or another, so I'm afraid I can't put it on GitHub or GitLab etc.

Output of `retype watch` and `retype build` (output is identical for both commands):
![image](https://user-images.githubusercontent.com/78737454/178969738-b2168dd2-6d5c-4447-9c3c-0325b5ce492f.png)

Output of `retype watch --verbose` is also completely identical.

I never get the usual output that includes the number of pages built, so I'm a bit lost as to what's causing this.

**Reply by WitekioPhil** [reply]
Sure!

1. No
2. No
3. Yes, see below:

![image](https://user-images.githubusercontent.com/78737454/178970159-d1df0360-bb86-4299-a2ff-9f676372d888.png)

Led me to check version, which was 1.7.0.

I tried to update using `npm update retypeapp --global`, but version remains the same at 1.7.0.

Could this issue be due to using an older version?

**Reply by geoffreymcgill** [reply]
okay, I think I might know why the error is thrown. You might be using a version of Retype that is now greater than one year old. 

Please try updating to a newer release, or ideally the latest release.

https://retype.com/guides/getting-started/#update

In the source code, I see the following comment:

```
// Only 1 Year from the release date is allowed without a license key
```

Greater than one year, Retype throws the error that you are seeing. Obviously that error msg is not descriptive enough. I am going to review and improve that message, plus determine whether that rule is even required. 

Updating to the latest release should fix the issue.

**Reply by geoffreymcgill** [reply]
> I tried to update using npm update retypeapp --global, but version remains the same at 1.7.0.

Please try to [uninstall](https://retype.com/guides/getting-started/#uninstall), then run a fresh [install](https://retype.com/guides/getting-started/#install).

If you originally installed using NPM, the two commands to run would be:

```
npm uninstall retypeapp --global
npm install retypeapp --global
```

Each of those should only take a few seconds total to execute. 

Hope this helps.

**Reply by WitekioPhil** [reply]
Thanks @geoffreymcgill.

I think I have a nodejs/npm issue preventing me from uninstalling.

After executing the uninstall command I still get 1.7.0 in response to `retype --version`
(both globally and locally, and restarting system not just the shell)

Off to solve that problem, I'll then post on here with results.

**Reply by geoffreymcgill** [reply]
Maybe you installed using Yarn? You could try running the following commands:

```
yarn global remove retypeapp
yarn global add retypeapp
```

If there is any chance you (or as reference for anyone reading this in the future) installed using .NET, run the following commands to uninstall and re-install the latest:

```
dotnet tool uninstall retypeapp --global
yarn global add retypeapp
```

More details and options at https://retype.com/guides/getting-started/.

Hope this helps.

**Reply by WitekioPhil** [reply]
Thanks again @geoffreymcgill!

Node / NPM had got in a state on my dev machine, uninstalled both, manually removed any remaining files, then a fresh install and I now have retype 2.3.0, and my notebook works again.

Sounds like you were correct about `1 year from release date` restriction. Is this documented anywhere public?

**Reply by geoffreymcgill** [reply]
> Is this documented anywhere public?

No. I am review the history of that validation rule and whether it's still required. Certainly the error msg will be improved.

**Reply by WitekioPhil** [reply]
Great, many thanks for your help @geoffreymcgill!
